src/Forms/UrlField.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace SilverStripe\Forms;
+
+/**
+ * Text input field with validation for a url using filter_var($url, FILTER_VALIDATE_URL)
+ * The value of the field must start with a protocol such as http:// or https://
+ */
+class UrlField extends TextField
+{
+    protected $inputType = 'text';
+
+    public function Type()
+    {
+        return 'text';
+    }
+
+    /**
+     * @param Validator $validator
+     *
+     * @return string
+     */
+    public function validate($validator)
+    {
+        $result = true;
+        $this->value = trim($this->value ?? '');
+        // filter_var() is recommended by https://urlregex.com/
+        if ($this->value && !filter_var($this->value, FILTER_VALIDATE_URL)) {
```

Comment:
`filter_var` with `FILTER_VALIDATE_URL` uses [RFC2396](http://www.faqs.org/rfcs/rfc2396.html) which is obsolete.
`parse_url()` uses [RFC3986](http://www.faqs.org/rfcs/rfc3986.html) which is what is currently in effect.

We've run into this problem before: https://github.com/silverstripe/silverstripe-tx-translator/pull/10#discussion_r1187024615

Please use `parse_url()` or something else that supports the current standard, and add some more robust tests to validate that.

Reply:
Updated